stock_request/models/stock_request.py

```diff
@@ -0,0 +1,301 @@
+        states={'draft': [('readonly', False)]},
+        readonly=True, required=True,
+        help="Quantity, specified in the unit of measure indicated in the "
+             "request.",
+    )
+    product_qty = fields.Float(
+        'Real Quantity', compute='_compute_product_qty',
+        digits=0, store=True, readonly=True,
+        help='Quantity in the default UoM of the product',
+    )
+    procurement_group_id = fields.Many2one(
+        'procurement.group', 'Procurement Group', copy=False, readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Moves created through this stock request will be put in this "
+             "procurement group. If none is given, the moves generated by "
+             "procurement rules will be grouped into one big picking.",
+    )
+    company_id = fields.Many2one(
+        'res.company', 'Company', required=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self.env['res.company']._company_default_get(
+            'stock.request'),
+    )
+    expected_date = fields.Datetime(
+        'Expected date', default=fields.Datetime.now, index=True,
+        required=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Date when you expect to receive the goods.",
+    )
+    picking_policy = fields.Selection([
+        ('direct', 'Receive each product when available'),
+        ('one', 'Receive all products at once')],
+        string='Shipping Policy', required=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        default='direct',
+    )
+    move_ids = fields.One2many('stock.move', 'stock_request_id',
+                               string='Stock Moves', readonly=True,
+                               )
+    picking_ids = fields.One2many('stock.picking',
+                                  compute='_compute_picking_ids',
+                                  string='Pickings', readonly=True,
+                                  )
+    qty_in_progress = fields.Float(
+        'Qty In Progress', digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute='_compute_qty', store=True,
+        help="Quantity in progress.",
+    )
+    qty_done = fields.Float(
+        'Qty Done', digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute='_compute_qty', store=True,
+        help="Quantity completed",
+    )
+    picking_count = fields.Integer(string='Delivery Orders',
+                                   compute='_compute_picking_ids',
+                                   readonly=True,
+                                   )
+    route_id = fields.Many2one('stock.location.route', string='Route',
+                               readonly=True,
+                               states={'draft': [('readonly', False)]},
+                               ondelete='restrict')
+
+    @api.depends('picking_ids')
```

Comment:
@jbeficent I think this should be move_ids.

Then, in the compute, don't do a search but use request.move_ids.filtered(lambda m: m.state '= 'cancel').mapped('picking_ids')